InformationScripting/scripts/color.py

```diff
@@ -8,6 +8,6 @@
 for ts in inputs:
     tuples = ts.take("ast")
     for tuple in tuples:
-        tuple.add(NamedProperty("color", color))
-        ts.add(tuple)
+        t = Tuple([NamedProperty("color", color), NamedProperty("ast", tuple.ast)])
```

Comment:
Hmm, using tuples from python sounds appealing. What is the difference in type safety regarding these two different approaches?